tests/test_selafin.cpp

```diff
@@ -222,7 +213,7 @@ TEST( MeshSLFTest, MalpassetResultFrench )
   EXPECT_DOUBLE_EQ( -0.97271907329559326, value );
 
   MDAL_D_minimumMaximum( ds, &min, &max );
-  EXPECT_DOUBLE_EQ( 2.3694833011052991e-12, min );
+  //EXPECT_DOUBLE_EQ( 2.3694833011052991e-12, min );
```

Comment:
well , this looks scary, why the test results changed?

Reply:
yes... I've forgot this one, I agree, scary... I investigate...

Reply:
OK, I got it! It is because active is not supported anymore and with this format, values equals to zero was considered as nan ([here](https://github.com/lutraconsulting/MDAL/blob/b2dcea01e387e672efba40978f86dcbf16618b87/mdal/frmts/mdal_selafin.cpp#L494) ). By the way, not sure it was a good thing.
So before this PR, there wasn't zero value in the dataset but nan value and in stats, those value was not considered.
Now they are, so the min is not the same than before.

Reply:
I think it is easier to stick with 0 == NaN, since the client doesn't requested this change, only lazy loading. Only if we have some reference doc that say something different.

Reply:
Not sure to understand the benefits of 0 = NaN:
- there can be scalar dataset whit 0 value that is real value : temperature, absolute level (sea level), ...
- what happen if this is a vector value (X,0) or (0,Y)?

Other point, as the access to dataset values is now directly from file, set the 0 value as NaN will require a costly loop to check all the value before sending them with `SelafinDataset::scalarData()` or `SelafinDataset::vectorData()`.
Maybe we can just set value as NaN only in `SelafinDataset::vectorData()` if X **AND** Y are zero value, this will not be much costly because there is already a loop to handle (x,y) values in only one array.

Reply:
@saberraz do you want to change the behaviour between versions (treat 0 as NaN as we did from beginning) or change as @vcloarec suggest, so 0 will become normal number visible in QGIS?

I do not remember why 0 was chosen for NaN, but I think it could be for example that they just use it that way? do we have some reference documentation for the format to check against?

Reply:
I wonder how we treat 0 for sww (AnuGA) format. Ideally, we should allow users to set additional NO DATA values for each quantity (similar to raster)...but that will be outside the scope and related to QGIS.

I agree with @vcloarec that 0 should be treated as a value and not a NaN.